proofs/tree.go

```diff
@@ -470,26 +624,93 @@ func FlattenMessage(message, messageSalts proto.Message) (nodes [][]byte, propOr
 	return f.nodes, f.propOrder, nil
 }
 
-// getStringValueByProperty gets a value from a struct and returns the value. This method does not yet
-// support nested structs. It converts the value to a string representation.
+// normalizeDottedProperty performs camel case conversion without removing slice characters ([])
+func normalizeDottedProperty(prop string) string {
+	arr := strings.Split(prop, ".")
+	for idx, key := range arr {
+		var propAux string
+		re := regexp.MustCompile(`(.*)(\[(.*)])`)
+		if re.MatchString(key) {
+			prefix := re.ReplaceAllString(key, "$1")
+			strIdx := re.ReplaceAllString(key, "$3")
+			propAux = fmt.Sprintf("%s[%s]", strcase.ToCamel(prefix), strIdx)
+		} else {
+			propAux = strcase.ToCamel(key)
+		}
+
+		if idx == 0 {
+			prop = fmt.Sprintf("%s", propAux)
+		} else {
+			prop = fmt.Sprintf("%s.%s", prop, propAux)
+		}
+	}
+	return prop
+}
+
+// getDottedValueByProperty takes a dotted property and retrieves the interface value of an object
+// It supports nested fields and slices
+func getDottedValueByProperty(prop string, value interface{}) (interface{}, error) {
+	prop = normalizeDottedProperty(prop)
+	if reflect.TypeOf(value).Kind() != reflect.Struct && reflect.TypeOf(value).Kind() != reflect.Ptr {
+		return nil, errors.New("non-struct interface not supported")
+	}
+	arr := strings.Split(prop, ".")
+	value = DereferencePointer(value)
+	var err error
+	for _, key := range arr {
+		re := regexp.MustCompile(`(.*)(\[(.*)])`)
+		if re.MatchString(key) {
+			prefix := re.ReplaceAllString(key, "$1")
+			strIdx := re.ReplaceAllString(key, "$3")
+			idx, err := strconv.Atoi(strIdx)
+			if err != nil {
+				return nil, err
+			}
+			value, err = GetFieldOfStruct(value, prefix)
+			if err != nil {
+				return nil, err
+			}
+			if value == nil {
+				return nil, nil
+			}
+			if reflect.ValueOf(value).Len() <= idx {
+				return nil, errors.New("Index out of bounds for slice element")
+			}
+			value = reflect.ValueOf(value).Index(idx).Interface()
+		} else {
```

Comment:
else can be removed here and move the block out ?

Reply:
I will move the value == nil out of the if blocks